app/src/main/java/org/gdg/frisbee/android/appwidget/UpcomingEventWidgetProvider.java

```diff
@@ -123,7 +123,18 @@ public void onNotFound(String key) {
                 });
 
         }
-
+        private Event getNextEvent(List<Event> listEvents) {
+            if (listEvents == null || listEvents.size() == 0) {
+                return null;
+            }
+            Event ret = listEvents.get(0);
```

Comment:
Maybe, I did not understood what the getChapterEvenList(DateTime start, DateTime finish) method of the backend does. my assumption was that it returns all the events between start and finish. new Datetime() is set to the current system millisecond in the default time zone(that is the equivalent of now I think). `App.getInstance().getGdgXHub() .getChapterEventList(homeGdg.getGplusId(), new DateTime(), new DateTime().plusMonths(1))` is therefore going to retrieve events between today and today + 1 month. From this list, it is either we loop through to get the closest date or if the list is sorted it will be the first element. I don't think it is necessary to separate the list in the backend. It will be very useful if we can confirm that the list from the backend is sorted and please do correct me if I have a wrong assumption on how the getChapterEventList works